base-files: fix nand_do_platform_check fail

ptpt52 opened the pull request:
This change ensures compatibility with both types of sysupgrade-tar files.

1. For some boards like xiaomi,redmi-router-ax6s, sysupgrade-tar is pack in directory `vendor,name/`
2. For some boards like xiaomi,mi-router-3g, sysupgrade-tar is pack in directory `vendor_name/`

And ensure that the platform_check_image() function returns an error code.

graysky2 reviewed (approved):
Worked for me in combo with https://github.com/openwrt/openwrt/pull/11115 / please merge

ptpt52 committed:
base-files: fix nand_do_platform_check fail

This change ensures compatibility with both types of sysupgrade-tar files.

1. For some boards like xiaomi,redmi-router-ax6s, sysupgrade-tar
   is pack in directory `vendor,name/`

2. For some boards like xiaomi,mi-router-3g, sysupgrade-tar is pack
   in directory `vendor_name/`

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>

ptpt52 committed:
mediatek: fix wrong return code in platform_check_image()

Ensure that the platform_check_image() function returns an error code.

Signed-off-by: Chen Minqiang <ptpt52@gmail.com>

ptpt52 commented:
ping @dangowrt @981213

ptpt52 commented:
please re-review, I think my update fix some wrong return code

ptpt52 commented:
merged.
